conda_self/query.py

```diff
@@ -73,15 +84,11 @@ def permanent_dependencies() -> set[str]:
     prefix_graph = PrefixGraph(installed)
 
     protect = [*PERMANENT_PACKAGES]
-    for record in installed:
-        with suppress(NoDistInfoDirFound):
-            for pkg_info in PackageInfo.from_record(record):
-                if "conda" in pkg_info.entry_points():
-                    protect.append(record.name)
-                    break
+    if add_plugins:
+        protect.extend(_find_plugins_in_record_list(installed))
```

Comment:
Let's keep this inline since I don't want to expose this as an API for modification via monkeypatching.

Reply:
I re-added the inline code.